config/jobs/garden-shoot-trust-configurator/garden-shoot-trust-configurator-build-images.yaml

```diff
@@ -0,0 +1,48 @@
+postsubmits:
+  gardener/garden-shoot-trust-configurator:
+  - name: post-garden-shoot-trust-configurator-build-images
+    cluster: gardener-prow-trusted
+    skip_if_only_changed: '^VERSION$'
+    branches:
+    - ^main$
+    annotations:
+      description: Garden Shoot Trust Configurator image build on main branch
+      testgrid-dashboards: garden-shoot-trust-configurator
+      testgrid-days-of-results: "60"
+    decorate: true
+    max_concurrency: 1
+    spec:
+      serviceAccountName: image-builder
+      containers:
+      - name: image-builder
+        image: europe-docker.pkg.dev/gardener-project/releases/ci-infra/image-builder:v20250825-738a51e
+        command:
+        - /image-builder
+        args:
+        - --log-level=info
+        - --docker-config-secret=gardener-prow-gcr-docker-config
+        - --registry=europe-docker.pkg.dev/gardener-project/snapshots/gardener
+        - --cache-registry=europe-docker.pkg.dev/gardener-project/snapshots/ci-infra/kaniko-cache
+        - --target=garden-shoot-trust-configurator
+        - --add-version-tag=true
+        - --add-version-sha-tag=true
+        - --add-fixed-tag=latest
+        - --kaniko-image=gcr.io/kaniko-project/executor:v1.24.0
+        # image-builder is the pod which is "scheduled" to a node. The pods created by image-builder have an affinity rule
+        # which schedules them to the same node as their parent image-builder. This needs to be done, that PVCs could be mounted
+        # to multiple build pods in parallel.
+        # For a proper scheduling the combined resource requests of all build pods are assigned to this pod, even though it does not
+        # use them. The resource requests of build pods themselves are "0"
+        resources:
+          requests:
+            cpu: 6
+            memory: 2Gi
+      # Node selector is copied to build pods
+      nodeSelector:
+        dedicated: high-cpu
+      # Tolerations are copied to build pods
+      tolerations:
+      - key: "dedicated"
+        operator: "Equal"
+        value: "high-cpu"
+        effect: "NoSchedule"
```

Comment:
Do we really need this image build job?
There is a Github workflow which builds the image too ([ref](https://github.com/gardener/garden-shoot-trust-configurator/blob/de0ce800a0de77088392fab5e678efb9d9b816bf/.github/workflows/build.yaml#L26-L60)).

Reply:
There is indeed an overlap - both this prow job and the GitHub Action build the same garden-shoot-trust-configurator OCI image and both push to a registry with latest and version-specific tags. Will remove the prow job here. Thank you for catching this!